### ISSUE TITLE
Any chance there will be a CAM simulator for this FoamCut

### ISSUE THREAD

**Issue by luzpaz** [Shkolik/Foamcut#1]
I know it's a big ask but would there be a possibility to create a CAM simulator (similar to how the current one works re-rewritten recently by @shaise) that supports simulating FoamCut gcode ?

**Comment by shaise** [comment]
This will be quite hard, as the gcode is not a simple 2.5 dimension CNC.

**Comment by Shkolik** [comment]
It's a really huge task. Maybe in future I'll tackle it, but for now there is 2 workarounds:
1) each path has property on a view tab named "Show projection lines" - it will connect ends of opposite paths with a line, so you can examine where and how wire will go. I also have plans to add similar functionality to Route and show this lines with given space along whole route.

![image](https://github.com/user-attachments/assets/fe6a5264-8a7e-43e8-bffe-8ea84ea725f8)

2) You can use "DevSim CNC Foam" to check gcode. It's free and all you need to do is setup machine in a sim the same way as your real machine:

![image](https://github.com/user-attachments/assets/8e8f4b78-97b3-4999-a26a-03b5c550b060)

**Comment by luzpaz** [comment]
Thank you for response. Is it possible for your to show No. 1 in documentation ? Perhaps show a screencast (can it be automated to animate via python at all?)

Regarding No. 2, DevSim CNC Foam, is there a way to auto load it from within FreeCAD or trigger it from within FreeCAD to open and run a simulation ?

**Comment by Shkolik** [comment]
1) I'm working on some screencast examples right now. Just a bit too busy - it's flying season still. I'll finish with examples/documentation by the end of November.

2) Do not think it's possible. I mean it's possible to run process, but free version of simulator doesn't support console arguments. So even start it with gcode loaded not possible.

**Comment by luzpaz** [comment]
> but free version of simulator doesn't support console arguments. So even start it with gcode loaded not possible.  

Are there any FLOSS options? If not, perhaps it would be cool to find other enthusiasts that may be open to collaborating?

**Comment by Shkolik** [comment]
Yep, that would be great if someone will jump into this project since I'm not aware of any FLOSS alternatives. Hope this topic will gain some interest in a future. For now it's only me and rarely Andrei helping with some ideas. It's just a start so will see...

**Comment by luzpaz** [comment]
@chennes & @yorikvanhavre perhaps this feature is worth suggesting an FPA grant for?

**Comment by luzpaz** [comment]
Maybe worth keeping an eye on https://github.com/ZviGoldenberg/FreeCAD_Simulator ?

**Comment by Shkolik** [comment]
> Maybe worth keeping an eye on https://github.com/ZviGoldenberg/FreeCAD_Simulator ? (FreeCAD simulator for ACS 5-axis stage)

The main problem of making sim is not number of axis, but toolbit in normal CNC has shape and size that not changing during cut but when you cutting foam it's changing and depends on speed of each individual pass on different parts of stock. It's probably not hard to implement just following gcode with a fixed kerf, but it will be not accurate. I think DevSim is doing exactly that...

**Comment by luzpaz** [comment]
> 1. I'm working on some screencast examples right now. Just a bit too busy - it's flying season still. I'll finish with examples/documentation by the end of November.

Did this every come to fruition ?

**Comment by Shkolik** [comment]
> Did this every come to fruition ?

Unfortunately not yet. Cannot do it at works since we returned to the office, and cannot do it at night from home... Will try to make it happen on weekend when will have free time from household projects. I have done 2 videos but do not like how they came out, so will need to redo them as well.

**Comment by Shkolik** [comment]
Nope, just a bug fixes. Extremely busy and don't see any more or less free time in the near future.
